### PR TITLE
CICD: Update docker containers to newer versions

### PR DESCRIPTION
**Pull request by onetechnical**
## Summary

Update containers to newer supported versions.

## Test Plan

Run build pipeline successfully.

### CODE REVIEW

**Review by algobarb — DISMISSED** [review]

**Review by algobarb — APPROVED** [review]
We should probably update Algorand docs here too with the new ubuntu version: https://github.com/algorand/docs/blob/8764866bc500d521fa131cd4937f7339909c6018/docs/run-a-node/setup/install.md?plain=1#L58

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:18.04 as builder
+FROM ubuntu:22.04 as builder
```
Comment (winder) [inline Dockerfile:1]: You could also update bullseye to `bookworm` or `bookworm-20230703-slim`

```
FROM debian:bullseye-slim as final
```

Comment (onetechnical) [reply Dockerfile:1]: Ooh, let's do that in a follow-up; I don't want to block this release, and want to think about testing that (but I don't anticipate any problems with it).

Comment (onetechnical) [reply Dockerfile:1]: Well, a simple build test worked, I'll go with it.

**Review by winder — APPROVED** [review]

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**